autoscheduler/frontend/src/components/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/ExpandedCourseCard.tsx

```diff
@@ -48,15 +51,21 @@ const ExpandedCourseCard: React.FC<ExpandedCourseCardProps> = ({ onCollapse, id
   return (
     <Card>
       <div className={styles.header}>
-        <div className={styles.headerGroup}>
+        <div
+          className={styles.headerGroup}
+          onClick={(): void => onRemove(id)}
```

Comment:
When this button is clicked, `ExpandedCourseCard` will actually call the `onRemove` function given to it in its props, which is actually an anonymous function in `CourseSelectCard.tsx`. That anonymous function then calls the `onRemove` function given to it in *its* props, which is actually an anonymous function in `CoursesSelectColumn.tsx`. This is actually a design pattern in React, and it's called "prop drilling," because the actual function we want to call in `CourseSelectCard` is separated from the event handler in `ExpandedCourseSelectCard`, so we drill it through `CourseSelectCard` in the form of props.

However, in this case, we are not actually piercing the heavens with our drill, as this drilling is rather unnecessary. It makes more sense to just move the `dispatch` call into `ExpandedCourseCard`, which gets rid of all of the unneccesary `onRemove` props.